bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -797,7 +849,7 @@ public int getReadOffset()
      * @return boolean
      */
     @Override
-    public synchronized boolean hasNext()
```

Comment:
You are right - the object belongs to single subscriber or publisher or control request. I don't recall why it was synchronized, probably remnant of refactoring from earlier incarnation of BS.